core/src/main/java/hudson/diagnosis/ReverseProxySetupMonitor.java

```diff
@@ -62,26 +67,35 @@ public boolean isActivated() {
     }
 
     public HttpResponse doTest() {
-        String referer = Stapler.getCurrentRequest().getReferer();
         Jenkins j = Jenkins.getInstance();
+        String referer = Stapler.getCurrentRequest().getReferer();
         // May need to send an absolute URL, since handling of HttpRedirect with a relative URL does not currently honor X-Forwarded-Proto/Port at all.
-        String redirect = j.getRootUrl() + "administrativeMonitor/" + id + "/testForReverseProxySetup/" + (referer != null ? Util.rawEncode(referer) : "NO-REFERER") + "/";
+        
+        String redirect = buildRedirect(referer, j);
```

Comment:
In the case of a referrer: 

`10.10.10.10/some/path/`

In the case of no referrer:
`NO-REFERRER`

Now it's:

Referrer: 

`10.10.10.10/some/path/~~no%2Fscheme~~`
No Referrer:
`NO-REFERRER/~~no%2Fscheme~~` // note that an encoded portion is always passed--this is the case that changed and introduced the regression